core/src/main/scala/BinaryTree.scala

```diff
@@ -0,0 +1,250 @@
+/**
+ * Created by Nicolas A Perez (@anicolaspp) on 1/29/16.
+ */
+
+package dogs
+
+import Predef._
+import dogs.Order.{GT, EQ, LT}
+import scala.annotation.tailrec
+import scala.math
+
+
+
+abstract class BinaryTree[A] {
+
+  def value: Option[A]
+
+  val left: BinaryTree[A]
+  val right: BinaryTree[A]
```

Comment:
what is the usefulness of these to someone? are these two methods just useful in order to write tests?

seems like we don't want someone actually trying to traverse the tree themselves, we just want to give them combinators that work according to our invariants, that is that they will work in order, and that they will have a predictable computational complexity